toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -1289,16 +1471,86 @@ func (b *LiveOSIsoBuilder) createImageFromUnchangedOS(baseConfigPath string, iso
 		return fmt.Errorf("failed to convert iso configuration to isomaker configuration format:\n%w", err)
 	}
 
-	err = b.updateGrubCfg(b.artifacts.savedKernelArgsFilePath, b.artifacts.grubCfgPath, extraCommandLine)
+	pxeIsoImageUrl := ""
+	if pxeConfig != nil {
+		pxeIsoImageUrl = pxeConfig.IsoImageUrl
+	}
+
+	err = b.updateGrubCfg(b.artifacts.savedConfigsFilePath, b.artifacts.isoGrubCfgPath, extraCommandLine, b.artifacts.pxeGrubCfgPath, pxeIsoImageUrl, outputImageBase)
 	if err != nil {
 		return fmt.Errorf("failed to update grub.cfg:\n%w", err)
 	}
 
-	err = b.createIsoImage(additionalIsoFiles, outputImageDir, outputImageBase)
+	isoImagePath, err := b.createIsoImage(additionalIsoFiles, outputImageDir, outputImageBase)
 	if err != nil {
 		return fmt.Errorf("failed to create iso image:\n%w", err)
 	}
 
+	err = populatePXEArtifactsDir(isoImagePath, b.workingDirs.isoBuildDir, outputPXEArtifactsDir, outputImageBase)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// populatePXEArtifactsDir
+//
+//   - This function takes in an liveos iso, and extracts its artifacts unto a
+//     folder for easier copying to a PXE server later by the user.
+//   - It also renames the liveos iso grub-pxe.cfg to grub.cfg.
+//
+// inputs:
+//
+//   - 'isoImagePath':
+//     path to a liveos iso image.
+//   - 'buildDir'
+//     path to a directory to hold intermediate files.
+//   - 'outputPXEArtifactsDir'
+//     path to the output directory where the extract artifacts will be saved to.
+//   - 'outputImageBase':
+//     base name of the image to generate. The generated name will be on the
+//     form: {outputImageDir}/{outputImageBase}.iso
+//
+// outputs:
+//
+//   - creates a folder with PXE artifacts.
+func populatePXEArtifactsDir(isoImagePath string, buildDir string, outputPXEArtifactsDir string, outputImageBase string) error {
+
+	if outputPXEArtifactsDir == "" {
+		return nil
+	}
+
+	logger.Log.Infof("Copying PXE artifacts to (%s)", outputPXEArtifactsDir)
+
+	err := os.RemoveAll(outputPXEArtifactsDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove (%s):\n%w", outputPXEArtifactsDir, err)
+	}
+
+	err = extractIsoImageContents(buildDir, isoImagePath, outputPXEArtifactsDir)
+	if err != nil {
+		return err
+	}
+
+	isoGrubCfg := filepath.Join(outputPXEArtifactsDir, "boot/grub2/grub.cfg")
+	pxeGrubCfg := filepath.Join(outputPXEArtifactsDir, "boot/grub2/grub-pxe.cfg")
```

Comment:
You should place these 2 file paths in constants.

Reply:
fixed.